salt: Lower Prometheus rule alert disk thresholds

alexandre-allard opened the pull request:
**Component**: salt, prometheus

**Context**: 
When disk space left is less than 10%, kubelet starts to evict pods to try to make some space.
The issue is our monitoring thresholds are higher than the ones from kubelet, so we never receive
alerts before a disk is actually full.

**Summary**:
We lower the thresholds for the following alerts

NodeFilesystemAlmostOutOfSpace:
  - warning from 5% to 20%
  - critical from 3% to 12%

NodeFilesystemAlmostOutOfFiles:
  - warning from 5% to 15%
  - critical from 3% to 8%

**Acceptance criteria**: 

---

Closes: #3474

alexandre-allard committed:
salt: Lower Prometheus rule alert disk thresholds

We lower the thresholds for the following alerts

NodeFilesystemAlmostOutOfSpace:
  - warning from 5% to 20%
  - critical from 3% to 12%
NodeFilesystemAlmostOutOfFiles:
  - warning from 5% to 15%
  - critical from 3% to 8%

Otherwise we don't receive alert before kubelet
starts evicting pods when disk is full as its
threshold is set to 10% of available disk space
and 5% of inodes free.

Refs: #3474

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/3474-lower-disk-alert-thresholds` into
`development/2.10`
* [w/2.11/bugfix/3474-lower-disk-alert-thresholds](https://github.com/scality/metalk8s/compare/development/2.11...w/2.11/bugfix/3474-lower-disk-alert-thresholds) will be merged into `development/2.11` 



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @teddyandrieux

* @alexandre-allard-scality

* @jbwatenbergscality

* @lucieleonard

* @chengyanjin

* @gdemonet

* @thomasdanan

alexandre-allard commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/fd59c2d8fb5e3d43571c20b603b8aeef484e425c) did not succeed in branch bugfix/3474-lower-disk-alert-thresholds.




*The following options are set:* **approve**

alexandre-allard committed:
tools: Update alert rules extract

```
./tools/rule_extractor/rule_extractor.py \
  -i <control-plane-ip> -p 8443 -t rules
```

Refs: #3474

alexandre-allard committed:
eve: Fix wait_debug step to handle numeric chars

The `debug` could only be composed of alphabetic
characters and dashes, but we are now putting the
dist version in the stage name (e.g. centos-7).

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @teddyandrieux

* @alexandre-allard-scality

* @jbwatenbergscality

* @lucieleonard

* @chengyanjin

* @gdemonet

* @thomasdanan










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/d2ab58e9df1d40d58c3f7b986e9e17431121a9c5) did not succeed in branch w/2.11/bugfix/3474-lower-disk-alert-thresholds.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.10`

* :heavy_check_mark: `development/2.11`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.10`

* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
